examples/prompt-tuning-personachat.ipynb

```diff
@@ -274,30 +274,32 @@
     "TEMPERATURE = 0.6\n",
     "dialog = \"\"\n",
     "\n",
-    "while True:\n",
-    "    user_phrase = input()\n",
-    "    if len(user_phrase) == 0:\n",
-    "        break\n",
-    "    dialog += f\"{user_phrase}\\n-----\\n\"\n",
-    "    inputs = tokenizer([dialog], return_tensors='pt')['input_ids']\n",
-    "    outputs = model.generate(\n",
-    "        inputs,\n",
-    "        temperature=TEMPERATURE,\n",
-    "        do_sample=True,\n",
-    "        top_k=TOP_K,\n",
-    "        eos_token_id=tokenizer.eos_token_id,\n",
-    "        max_new_tokens=MAX_TOKENS,\n",
-    "    )\n",
-    "    bloom_answer = tokenizer.batch_decode(outputs)[0]\n",
-    "    bloom_answer = bloom_answer[len(dialog):].split(\"\\n\")[0]\n",
-    "    print(bloom_answer)\n",
-    "    dialog += f\"{bloom_answer}\\n-----\\n\""
+    "with model.inference_session(max_length=512) as sess:\n",
+    "    while True:\n",
+    "        user_phrase = input()\n",
+    "        if len(user_phrase) == 0:\n",
+    "            break\n",
+    "        dialog += f\"{user_phrase}\\n-----\\n\"\n",
```

Comment:
When using `.generate()` reusing the inference session, you should not pass the entire prefix to `.generate()` every time. You should only pass the new user's phrase (the previous phrase and the model's output are already remembered in the attention caches).

This way, you don't need to store the entire history in the `dialog` variable. You can only store the last input/output in this example.

Also, I encourage you to show the tokens in real time, since waiting for ~10 sec for a new response is quite unpleasant. Please note how it's done in [my example](https://colab.research.google.com/drive/1Ervk6HPNS6AYVr3xVdQnY5a-TjjmLCdQ?usp=sharing).

Reply:
Thank you for your comment! I will change the behaviour of an interactive dialogue, and make it like in your example.